src/app/Archives/AllArchivedRecordingsTable.tsx

```diff
@@ -240,25 +240,34 @@ export const AllArchivedRecordingsTable: React.FunctionComponent<AllArchivedReco
     );
   };
 
-  const handleDeleteButton = () => {
+  const handleDeleteButton = React.useCallback(() => {
     if (context.settings.deletionDialogsEnabledFor(DeleteWarningType.DeleteArchivedRecordings)) {
       setWarningModalOpen(true);
     }
     else {
       handleDeleteRecordings();
     }
-  }
+  }, [handleDeleteRecordings])
+
+  const handleWarningModalClose = React.useCallback(() => {
+    setWarningModalOpen(false);
+  }, [setWarningModalOpen]);
+
+  const disableDeletionDialog = () => {
+    context.settings.setDeletionDialogsEnabledFor(DeleteWarningType.DeleteArchivedRecordings, false);
```

Comment:
Any particular reason to not make this one a `useCallback()` too?

Reply:
Yea, sorry, it "doesn't matter" in practical rendering terms - at least, given the rest of the structure of the component, at this point in time. It might be a slight case of premature optimization to apply `useCallback` here, but I think the penalty for it is small. I would rather err on the side of applying it when it's unnecessary and adding a little extra unneeded overhead, rather than not applying it and leaving it open to bugs in the future after this component is refactored or used in a new way.